src/coreclr/jit/gentree.cpp

```diff
@@ -16820,6 +16858,7 @@ bool GenTree::IsFieldAddr(Compiler* comp, GenTree** pBaseAddr, FieldSeqNode** pF
 
         *pBaseAddr = baseAddr;
         *pFldSeq   = fldSeq;
+        *pOffset   = offset;
         return true;
     }
```

Comment:
Just to check my understanding -- the invariant here will be that the sum of offsets saved in the field sequences equal `offset`? (Well, up to the subtraction you did above)